capellambse_context_diagrams/collectors/tree_view.py

```diff
@@ -51,13 +48,29 @@ def process_class(self, cls: ClassInfo, params: dict[str, t.Any]):
         self._process_box(cls.source, cls.partition, params)
 
         if not cls.primitive and isinstance(cls.target, information.Class):
+            assert cls.prop is not None
             self._process_box(cls.target, cls.partition, params)
-            edges = [
-                assoc
-                for assoc in self.all_associations
-                if cls.prop in assoc.navigable_members  # type: ignore[attr-defined]
-            ]
-            edge_id = edges[0].uuid
+
+            aggregation_kind_excludes = {
+                None,
+                modeltypes.AggregationKind.UNSET,
+            }
+            if cls.prop.association is not None:
+                edge_id = cls.prop.association.uuid
+            else:
+                logger.warning(
+                    "No Association found for %s set on 'navigable_members'",
+                    cls.prop._short_repr_(),
+                )
+                if cls.prop.kind not in aggregation_kind_excludes:
+                    styleclass = cls.prop.kind.name.capitalize()
+                else:
+                    styleclass = "Association"
+
+                i = self._edge_count.setdefault(cls.prop.uuid, 0)
+                i += 1
+                edge_id = f"__{styleclass}:{cls.prop.uuid}-{i}"
```

Comment:
Note that this doesn't increment the value stored in `self._edge_count`.

I think it'd be better to use a `collections.defaultdict(int)` here and simply do `self._edge_count[cls.prop.uuid] += 1`.